modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/ExtensionDefinitionParser.java

```diff
@@ -743,71 +743,90 @@ private boolean isExpressionFunction(MetadataType metadataType)
         }
     }
 
-    private ValueResolver parseDateTime(Object value, DateTimeType dataType, Object defaultValue)
+
+    private static ValueResolver doParseDate(Object value, Class<?> type)
     {
-        Class<?> type = getType(dataType);
-        if (isExpression(value, parser))
-        {
-            return new TypeSafeExpressionValueResolver((String) value, type);
-        }
 
-        if (type.equals(LocalDateTime.class))
+        if (value instanceof String)
         {
-            Date date = doParseDate(value, DATE_FORMAT, defaultValue);
-            return new StaticValueResolver(LocalDateTime.ofInstant(date.toInstant(), ZoneId.systemDefault()));
+            Object constructedValue = null;
+            DateTime dateTime = getParsedDateTime((String) value);
+
+            if (type.equals(LocalDate.class))
+            {
+                constructedValue = LocalDate.of(dateTime.getYear(), dateTime.getMonthOfYear(), dateTime.getDayOfMonth());
```

Comment:
I'm dropping the time part when the type I need to construct doesn't make use of that information (like Date for instance). The thing here is, that because LocalDate adapts the time to the local time zone, we cannot make assertions against the hour/date/minute. 

Looking through the code I saw that he have the same exact problem in [some other test](https://github.com/mulesoft/mule/blob/mule-4.x/modules/extensions-spring-support/src/test/java/org/mule/runtime/module/extension/internal/config/ParameterizedConfigParserTestCase.java#L237).
 
I think that, in order to make a complete test (including time zones) we should give support to types like [java.time.ZonedDateTime](https://docs.oracle.com/javase/8/docs/api/java/time/ZonedDateTime.html).